[WPE] WPE Platform: Use default values from WPESettings for WPEView

siwei-li opened the pull request:
#### 8d758321355e40d71444cd9c437deec32b32beed
<pre>
[WPE] WPE Platform: Use default values from WPESettings for WPEView
<a href="https://bugs.webkit.org/show_bug.cgi?id=284758">https://bugs.webkit.org/show_bug.cgi?id=284758</a>

Reviewed by Carlos Garcia Campos.

Set the size of the WPEView as well as the double click time and distance
with the values from WPESettings.

* Source/WebKit/WPEPlatform/wpe/WPESettings.cpp:
(_WPESettingsPrivate::_WPESettingsPrivate):
* Source/WebKit/WPEPlatform/wpe/WPESettings.h:
* Source/WebKit/WPEPlatform/wpe/WPEView.cpp:
(wpeViewConstructed):
(wpe_view_compute_press_count):

Canonical link: <a href="https://commits.webkit.org/288148@main">https://commits.webkit.org/288148@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8a8779db13f2611b65596649bb51ad0b47b71d59

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/81880 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/1406 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/35836 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/86437 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/32882 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/1441 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/9229 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/63866 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/21590 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/84950 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/1064 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/74516 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/44150 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/964 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/31331 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/72312 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/29302 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/87868 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/9122 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/6501 "Found 1 new test failure: imported/w3c/web-platform-tests/mediacapture-record/MediaRecorder-mimetype.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/72225 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/9307 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/70335 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/71448 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/15540 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/14465 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/496 "The change is no longer eligible for processing.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/12708 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/9073 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/14609 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/8912 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/12438 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/10720 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Source/WebKit/WPEPlatform/wpe/WPESettings.h:
```diff
@@ -162,7 +162,48 @@ typedef enum {
 #define WPE_SETTING_CURSOR_BLINK_TIME "/wpe-platform/cursor-blink-time"
-
+/**
+ * WPE_SETTING_TOPLEVEL_WIDTH:
```
I would call this `WPE_SETTING_TOPLEVEL_DEFAULT_WIDTH`

carlosgcampos commented on Source/WebKit/WPEPlatform/wpe/WPESettings.h:
```diff
@@ -162,7 +162,48 @@ typedef enum {
+/**
+ * WPE_SETTING_TOPLEVEL_WIDTH:
+ *
+ * The width of the toplevel size.
```
And we need to document that this is the default width of the toplevel used at construction.

carlosgcampos commented on Source/WebKit/WPEPlatform/wpe/WPESettings.h:
```diff
@@ -162,7 +162,48 @@ typedef enum {
+ */
+#define WPE_SETTING_TOPLEVEL_WIDTH "/wpe-platform/toplevel-size-width"
+/**
+ * WPE_SETTING_TOPLEVEL_HEIGHT:
```
I think we could have a single setting: `WPE_SETTINGS_TOPLEVEL_DEFAULT_SIZE` of type "uu"

carlosgcampos commented on Source/WebKit/WPEPlatform/wpe/WPESettings.h:
```diff
@@ -162,7 +162,48 @@ typedef enum {
+ *
+ * Default: 5
+ */
+#define WPE_SETTING_DOUBLE_CLICK_DISTANCE "/wpe-platform/double-click-distance"
```
```suggestion
#define WPE_SETTING_DOUBLE_CLICK_DISTANCE "/wpe-platform/events/double-click/distance"
```

carlosgcampos commented on Source/WebKit/WPEPlatform/wpe/WPESettings.h:
```diff
@@ -162,7 +162,48 @@ typedef enum {
+ *
+ * Default: 400
+ */
+#define WPE_SETTING_DOUBLE_CLICK_TIME "/wpe-platform/double-click-time"
```
```suggestion
#define WPE_SETTING_DOUBLE_CLICK_TIME "/wpe-platform/events/double-click/time"
```

carlosgcampos commented on Source/WebKit/WPEPlatform/wpe/WPESettings.h:
```diff
@@ -282,6 +323,5 @@ WPE_API gboolean     wpe_settings_set_boolean                     (WPESettings
                                                                    gboolean            value,
                                                                    WPESettingsSource   source,
                                                                    GError            **error);
-
```
No need to remove this line

carlosgcampos commented on Source/WebKit/WPEPlatform/wpe/WPEView.cpp:
```diff
@@ -172,9 +174,15 @@ static void wpeViewConstructed(GObject* object)
-    priv->height = 768;
+    auto* view = WPE_VIEW(object);
+    auto* priv = view->priv;
+    auto settings = wpe_display_get_settings(wpe_view_get_display(view));
```
```suggestion
    auto* settings = wpe_display_get_settings(priv->display.get());
```

carlosgcampos commented on Source/WebKit/WPEPlatform/wpe/WPEView.cpp:
```diff
@@ -172,9 +174,15 @@ static void wpeViewConstructed(GObject* object)
     G_OBJECT_CLASS(wpe_view_parent_class)->constructed(object);
 
     // FIXME: add API to set the default view size.
```
Remove this FIXME, it's what this patch fixes.

carlosgcampos commented on Source/WebKit/WPEPlatform/wpe/WPEView.cpp:
```diff
@@ -172,9 +174,15 @@ static void wpeViewConstructed(GObject* object)
+    auto width = wpe_settings_get_uint32(settings, WPE_SETTING_TOPLEVEL_WIDTH, &error.outPtr());
+    g_assert(!error.get());
+    priv->width = width;
+    priv->height = wpe_settings_get_uint32(settings, WPE_SETTING_TOPLEVEL_HEIGHT, &error.outPtr());
```
The getter only fails if the setting doesn't exist, and this is a builtin setting, so I think we can just ignore the error and pass nullptr.

carlosgcampos commented on Source/WebKit/WPEPlatform/wpe/WPEView.cpp:
```diff
@@ -954,8 +962,13 @@ guint wpe_view_compute_press_count(WPEView* view, gdouble x, gdouble y, guint bu
         // FIXME: make these configurable.
-        static const int doubleClickDistance = 5;
-        static const unsigned doubleClickTime = 400;
+        auto settings = wpe_display_get_settings(wpe_view_get_display(view));
```
```suggestion
        auto* settings = wpe_display_get_settings(priv->display.get());
```

carlosgcampos commented on Source/WebKit/WPEPlatform/wpe/WPEView.cpp:
```diff
@@ -954,8 +962,13 @@ guint wpe_view_compute_press_count(WPEView* view, gdouble x, gdouble y, guint bu
+        auto distance = wpe_settings_get_uint32(settings, WPE_SETTING_DOUBLE_CLICK_DISTANCE, &error.outPtr());
+        g_assert(!error.get());
+        int doubleClickDistance = distance;
+        unsigned doubleClickTime = wpe_settings_get_uint32(settings, WPE_SETTING_DOUBLE_CLICK_TIME, &error.outPtr());
```
Same about the error handling.

carlosgcampos commented on Source/WebKit/WPEPlatform/wpe/WPESettings.cpp:
```diff
@@ -555,6 +558,47 @@ WPE_SETTINGS_HELPER_API(const char*, string, G_VARIANT_TYPE_STRING, nullptr)
+ * @source: the source of the settings change
+ * @error: return location for a #GError or %NULL
+ *
+ * This is a simple wrapper around [method@WPESettings.set_value] that also validates
```
I don't think we want a wrapper for this, for complex types we can just use set_value/get_value

carlosgcampos commented on Source/WebKit/WPEPlatform/wpe/WPESettings.h:
```diff
@@ -162,7 +162,39 @@ typedef enum {
+ *
+ * Default: (1024, 768)
+ */
+#define WPE_SETTING_TOPLEVEL_DEFAULT_SIZE "/wpe-platform/toplevel-size"
```
```suggestion
#define WPE_SETTING_TOPLEVEL_DEFAULT_SIZE "/wpe-platform/toplevel-default-size"
```

carlosgcampos reviewed (approved):
Thanks!

siwei-li committed:
[WPE] WPE Platform: Use default values from WPESettings for WPEView
https://bugs.webkit.org/show_bug.cgi?id=284758

Reviewed by Carlos Garcia Campos.

Set the size of the WPEView as well as the double click time and distance
with the values from WPESettings.

* Source/WebKit/WPEPlatform/wpe/WPESettings.cpp:
(_WPESettingsPrivate::_WPESettingsPrivate):
* Source/WebKit/WPEPlatform/wpe/WPESettings.h:
* Source/WebKit/WPEPlatform/wpe/WPEView.cpp:
(wpeViewConstructed):
(wpe_view_compute_press_count):

Canonical link: https://commits.webkit.org/288148@main
